### ISSUE TITLE
About the CE loss

### ISSUE THREAD

**Issue by alice-cool** [bismex/HiCMD#5]
Dear scholar,
      Could you give me the link about the supplementary materials that was mentioned in your paper? And maybe the introduction of CE loss part is very brief. I want to ask you whether feature fi(from di com) is followed by a classifier to get the CE loss. And how to decide the number of nodes in the classifier? I guess the number of nodes in classifier maybe the number of ids in the training set such as 206 or 395.

**Comment by alice-cool** [comment]
Dear scholar,
In the attribute encoder in your figure, it includes a module named as GAP. I found in the paper"MUNIT_ECCV2018" the module includes global average pooling and a fc layer. Does your model only have global average pooling so you only name it as GAP?

**Comment by bismex** [comment]
Thank you for your interest.
This is the supplementary material link (https://openaccess.thecvf.com/content_CVPR_2020/supplemental/Choi_Hi-CMD_Hierarchical_Cross-Modality_CVPR_2020_supplemental.pdf).
You can check the detailed network structure in the supplementary material (page 3) and my code. 
I will reply to your questions.
1) The number of nodes in the classifier depends on the number of training IDs (RegDG:206, SYSU:395).
2) You can check the structure of the attribute encoder in the code 
(58 lines in https://github.com/bismex/HiCMD/blob/master/trainer.py, 
329-396 lines in https://github.com/bismex/HiCMD/blob/17e5f792ae203cabb9dea8cc0601f92ef9e1c387/reIDmodel.py#L329)
There is no additional fully connected layer.

Thanks,
Seokeon Choi

**Comment by alice-cool** [comment]
Dear scholar,
Thanks for your timely reply.

**Comment by alice-cool** [comment]
Dear scholar
I am very glad to read your paper. It is so cool and fantastic. I have a little confusion about the Part Adversarial loss about the difference of the two distribution about x1~Pdata(x1) and x1~p(x1).
And in your part KL divergence loss , you said in your paper "a prior Gaussian distribution" , does it refer to the mathematical concept "Central Limit Theory"?
If you have time, I hope you will help me.
Yours, sincerely.

**Comment by alice-cool** [comment]
Dear scholar,
I am very glad to read your paper. It is so cool and fantastic. I have a little confusion about the Generator. I think in your paper, all mathematics symbol G is the same module Generator, even though sometimes you call it as decoder.
The overall loss function is as follows:


L = L recon+ L kl +  L adv + Lce + L trip      (10)
And in your part Adversarial loss , you only referred to L adv. But I think in your paper, all function with G function is the Generator process. So I guess in the loss of L recon, which includes cross-modality reconstruction and the same modality reconstruction, also should be included in your Adversarial loss Section.
If you have time, I hope you will help me.
Yours, sincerely.

**Comment by bismex** [comment]
The answers are as follows:

1) I have a little confusion about the Generator. I think in your paper, all mathematics symbol G is the same module Generator, even though sometimes you call it as decoder.

-> We call our image generation module a generator, which consists of an encoder and a decoder. And, the symbol G denotes a decoder. We apologize for confusing your understanding.

2) The overall loss function is as follows: L = L recon+ L kl + L adv + Lce + L trip (10) And in your part Adversarial loss , you only referred to L adv. But I think in your paper, all function with G function is the Generator process. So I guess in the loss of L recon, which includes cross-modality reconstruction and the same modality reconstruction, also should be included in your Adversarial loss Section.

-> L_recon loss contains only the reconstruction losses (equations (1), (2), (3), and (4)). We explain the overall reconstruction loss in eq (5).

3) I have a little confusion about the Part Adversarial loss about the difference of the two distribution about x1~Pdata(x1) and x1~p(x1) .

-> There is no difference. It's a typo.

4) And in your part KL divergence loss , you said in your paper "a prior Gaussian distribution" , does it refer to the mathematical concept "Central Limit Theory" ?

-> We referred to Gaussian modeling of the prior distribution in GAN or other image translation papers.

**Comment by alice-cool** [comment]
Thanks for your reply.&nbsp;
Thanks all the time

---Original---
